docs/index.bs

```diff
@@ -3175,6 +3235,78 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
       1. Return |response|.
   </section>
 
+  <section algorithm>
+    <h3 id="verify-router-rule-algorithm"><dfn>VerifyRouterRule</dfn></h3>
+
+      : Input
+      :: |rule|, a {{RouterRule}}
+      :: |serviceWorker|, a [=/service worker=]
+      : Output
+      :: a boolean
+
+      1. If |rule|["{{RouterRule/condition}}"]["{{RouterCondition/urlPattern}}"] is the empty string, return false.
+      1. Let |rawPattern| be |rule|["{{RouterRule/condition}}"]["{{RouterCondition/urlPattern}}"].
+      1. If |rawPattern| is a [=string=], then:
+        1. Let |baseURL| be |serviceWorker|'s [=service worker/script url=].
+        1. Set |pattern| to the result of constructing a {{URLPattern}} using the {{URLPattern/URLPattern(input, baseURL)}} constructor steps given |rawPattern| and |baseURL|. If those steps throw, catch the exception and return false.
+      1. Otherwise, if |rawPattern| is {{URLPatternInit}}:
+        1. Set |pattern| to the result of constructing a {{URLPattern}} using the {{URLPattern/URLPattern(input)}} constructor steps given |rawPattern|. If those steps throw, catch the exception and return false.
+      1. Otherwise, return false.
+      1. If running the [=VerifyURLPattern=] algorithm with |pattern| returns false, return false.
+      1. Set |rule|["{{RouterRule/condition}}"]["{{RouterCondition/urlPattern}}"]'s associated {{URLPattern}} to |pattern|.
+      1. Return true.
+  </section>
+
+  <section algorithm>
+    <h3 id="verify-urlpattern-algorithm"><dfn>VerifyURLPattern</dfn></h3>
+
+    : Input
+    :: |pattern|, a {{URLPattern}}
+    : Output
+    :: a boolean
+
+    1. If running the [=VerifyURLPatternComponent=] algorithm with |pattern|'s [=urlpattern/protocol component=] returns false, return false.
```

Comment:
Let's wait until https://github.com/whatwg/urlpattern/pull/199 lands, and then we can get rid of this whole algorithm (and its sub-algorithm). Instead we can just say `|pattern| [=URLPattern/has regexp groups=]`.